src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PostMergeBlockProducer.cs

```diff
@@ -70,7 +70,21 @@ public Block PrepareEmptyBlock(BlockHeader parent, PayloadAttributes? payloadAtt
             // processing is only done here to apply block rewards in AuRa
             if (TrySetState(parent.StateRoot))
             {
-                block = ProcessPreparedBlock(block, null) ?? block;
+                var processedBlock = ProcessPreparedBlock(block, null);
+                if (processedBlock != null)
+                {
+                    block = processedBlock;
+                    if (Logger.IsDebug) Logger.Debug($"Processed empty block {block.Hash}");
+                }
+                else
+                {
+                    if (Logger.IsDebug) Logger.Debug($"Failed to process empty block {block.Hash}");
+                }
+                // block = ProcessPreparedBlock(block, null) ?? block;
```

Comment:
ok this now return not-null block where prepare empty block fails? Potential issue?

Reply:
only for AuRa, right? btw do we need TrySetState and ProcessPreparedBlock for mainnet? @jmederosalvarado @LukaszRozmej

Reply:
> ok this now return not-null block where prepare empty block fails? Potential issue?

@LukaszRozmej do you mean that even if `ProcessPreparedBlock` fails we would return a non-null block? What's the alternative then? Returning `null` here would make us fail block production, which seems worst to me.

Also, this behavior is already in master, it's not related to any changes from this PR.

Reply:
> only for AuRa, right? btw do we need TrySetState and ProcessPreparedBlock for mainnet? @jmederosalvarado @LukaszRozmej

Yes @MarekM25 I think you're right. Should we have it as an override in `AuRaMergePlugin`?

Reply:
Ok maybe:
1. Let's move those changes to `PostMergeAuRaBlockProducer` ?
2. if we return non-null bad block - unprocessed in AuRa it might be worse than not returning a block. Proposing a malicious block is worst than skipping a slot.